internal/store/kubernetesdiscoverys/reducers.go

```diff
@@ -50,10 +67,11 @@ func RefreshKubernetesResource(state *store.EngineState, name string) {
 	}
 
 	d := state.KubernetesDiscoverys[name]
-	r, err := k8sconv.NewKubernetesResource(d, aStatus)
+	filter, err := filterForResource(state, name)
```

Comment:
would it work to do an apicmp.DeepEqual() on both the discovery and the apply, and skip the whole refresh branch if they're the same?

this is supposed to be idempotent.

Reply:
In the problematic code path, [we already do](https://github.com/tilt-dev/tilt/blob/0f1dcaa3eec0e99e3b23883d6ff0546b1fc147ea/internal/store/kubernetesdiscoverys/reducers.go#L27-L32).

My (unverified) understanding of the issue is that there are a lot of pod status changes in the cluster, which generate KD status changes, which generate KD upsert actions, and since the KD status has changed, they generate calls to RefreshKubernetesResource, which reparse yaml that hasn't changed. I'll see if the user can confirm this.